app/modules/ads.js

```diff
@@ -552,6 +556,10 @@ class Ads {
 		}
 	}
 
+	waitForReady() {
```

Comment:
can we make public API more consistent?
`waitForReady()` which returns promise vs `onReady()` which accepts callback
Please decide on one and migrate the other.

Reply:
Isn't better to keep it optional or use same method pass callback and return Promise?

Reply:
personally, I don't like if you have too many paths to achieve the same. Then you need to debug every path separately. But if you believe it's better for you then let it be.